dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/ProgramRuleEngine.java

```diff
@@ -64,6 +68,11 @@
 {
     private static final String USER = "USER";
 
+    private static final String REGEX = "d2:inOrgUnitGroup\\( *(([\\d/\\*\\+\\-%\\. ]+)|" +
```

Comment:
This regex is for getting the entire function call with params. Maybe we should just do a simpler one to avoid any dependencies on future syntax changes? Perhaps even a simple string search is enough?

Reply:
I agree, this is why I removed it.
A simple string search is risky.
We need to find a performant way to get a set of orgUnits that will include the ones that we need